graph/services/category.service.go

```diff
@@ -0,0 +1,206 @@
+package services
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/Swejal08/go-ggqlen/graph/model"
+	"github.com/Swejal08/go-ggqlen/initializer"
+	"github.com/Swejal08/go-ggqlen/utils"
+	"github.com/doug-martin/goqu/v9"
+	"github.com/google/uuid"
+)
+
+var categoryFieldMapper = map[string]string{
+	"CategoryName": "category_name",
+}
+
+func CreateCategory(body model.NewCategory) (*model.Category, error) {
+	database := initializer.GetDB()
+
+	queryBuilder := initializer.GetQueryBuilder()
+
+	newId := uuid.New()
+
+	ds := queryBuilder.Insert("category").
+		Cols("id", "event_id", "category_name").
+		Vals(goqu.Vals{newId, body.EventID, body.CategoryName})
+
+	sql, _, err := ds.ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("An error occurred while generating the SQL", err.Error())
+	}
+
+	if _, err = database.Exec(sql); err != nil {
+
+		return nil, fmt.Errorf("An error occurred while executing the SQL", err.Error())
+	}
+
+	newCategory := &model.Category{
+		ID:           newId.String(),
+		EventID:      body.EventID,
+		CategoryName: body.CategoryName,
+	}
```

Comment:
Better to fetch this data from the SQL query itself using `RETURNING` clause rather than constructing manually.